File: uhrwerk-config/src/main/java/io/qimia/uhrwerk/config/representation/ParallelLoad.java
package io.qimia.uhrwerk.config.representation;


import io.qimia.uhrwerk.config.ConfigException;

public class ParallelLoad{

    private String query;
    private String column;
    private Integer num;

    public ParallelLoad() {}

    public String getQuery() {
        return query;
    }

    public void setQuery(String query) {
        this.query = query;
    }

    public String getColumn() {
        return column;
    }

    public void setColumn(String column) {
        this.column = column;
    }

    public Integer getNum() {
        return num;
    }

    public void setNum(Integer num) {
        this.num = num;
    }

    public void validate(String path){
        path += "parallel_load/";
        if(query == null){
            throw new ConfigException("Missing field: " + path + "query");
        }
        if(column == null){
            throw new ConfigException("Missing field: " + path + "column");
        }
        if(num == null){
            throw new ConfigException("Missing field: " + path + "num");
        }
    }

    @Override
    public String toString() {
        return "ParallelLoad{" +
                "query='" + query + '\'' +
                ", column='" + column + '\'' +
                ", num=" + num +
                '}';
    }
}

File: uhrwerk-config/src/main/kotlin/io/qimia/uhrwerk/config/representation/ParallelLoad.kt
package io.qimia.uhrwerk.config.representation

import io.qimia.uhrwerk.config.builders.ConfigException

data class ParallelLoad(
    var query: String? = null,
    var column: String? = null,
    var num: Int? = null
) {
    fun validate(path: String) {
        var path = path
        path += "parallel_load/"
        if (query == null) {
            throw ConfigException("Missing field: " + path + "query")
        }
        if (column == null) {
            throw ConfigException("Missing field: " + path + "column")
        }
        if (num == null) {
            throw ConfigException("Missing field: " + path + "num")
        }
    }
}